react/modules/react-setup-env/README.md

```diff
@@ -0,0 +1,61 @@
+# [React](https://github.com/rolling-scopes-school/tasks/tree/master/react) Setting Up Environment 🌟
+
+## Module Overview 📚
+
+This module introduces students to setting up a React development environment. It aims to provide a comprehensive
+understanding of how to install and configure the necessary tools and packages to start building a React application.
+The module covers a range of topics from installing Node.js and npm, to creating a new React application using Create
+React App, understanding the structure of a React project, and setting up Linter, Prettier, Husky, Webpack, and Vite.
+
+## Learning Objectives 🎯
+
+Students will:
+
+- Understand the prerequisites for setting up a React environment.
+- Learn how to install Node.js and npm.
+- Learn how to create a new React application using Create React App.
+- Get familiar with the structure of a React project.
+- Understand the purpose and setup process of Linter, Prettier, Husky, Webpack, and Vite.
+
+## Approximate Module Completion Time ⏱️
+
+- **[4 hours]**
+
+## Theory 📖
+
+Students are encouraged to study the following resources:
+
+1. **Prerequisites and Installing Node.js and npm:**
+   - [Node.js Official Website](https://nodejs.org/en/) - [15 minutes]
+   - [How to Install Node.js and npm](https://www.npmjs.com/get-npm) - [15 minutes]
+   - [npm vs npx](https://www.freecodecamp.org/news/npm-vs-npx-whats-the-difference/) — What’s the Difference? - [15 minutes]
+2. **Creating a New React Application and Understanding Project Structure:**
+   - [Start a New React Project](https://react.dev/learn/start-a-new-react-project) - [30 minutes]
+   - [React.js Architecture Pattern](https://www.knowledgehut.com/blog/web-development/react-js-architecture) - [30 minutes]
+3. **Setting Up Linter, Prettier, Husky, Webpack, and Vite:**
+   - [Configuration (new: eslint.config.js)](https://github.com/jsx-eslint/eslint-plugin-react?tab=readme-ov-file#configuration-new-eslintconfigjs)
+   - [ESLint and Prettier configuration for React project](https://dev.to/tsamaya/eslint-and-prettier-configuration-for-react-project-2gij) - [30 minutes]
+   - [Eslint Documentation](https://eslint.org/docs/latest/) - [30 minutes]
+   - [Prettier Documentation](https://prettier.io/docs/en/options.html) - [30 minutes]
+   - [Husky Guide](https://typicode.github.io/husky/guide.html) - [30 minutes]
+4. **Webpack and Vite:**
+   - [Understanding Webpack](https://webpack.js.org/guides/getting-started/) - [30 minutes]
+   - [Getting Started with Vite](https://vitejs.dev/guide/) - [30 minutes]
+
+[//]: # "TODO Practice?"
+
+## Practice 💻
+
+- Complete the task "Set Up Your First React Application".
+- Set up ESLint, Prettier, Husky in your React project.
+- Configure Webpack for your React application.
```

Comment:
We don't need to configure webpack in the Vite setup. However it might be important to explain old approaches and denote the difference (Vite uses esmodules)
Here is a link which explains details: https://vitejs.dev/guide/why